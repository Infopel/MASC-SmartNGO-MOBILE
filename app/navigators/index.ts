export * from "./root-navigator"
export * from "./navigation-utilities"
// export other navigators from here
export * from "./navigators.d"
export * from "./home-bottom-tab/home-bottom-tab-navigator"
export * from "./auth/auth-navigator"
export * from "./primary/primary-navigator"
export * from "./civic-incubator-details-tab/civic-incubator-details-tab-navigator"
export * from "./savings-group-details-tab/savings-group-details-tab-navigator"
