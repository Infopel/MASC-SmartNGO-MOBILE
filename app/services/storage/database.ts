import { MonitoringVisit } from './models/monitoring-visit';
import { Database } from '@nozbe/watermelondb';
import { Project } from './models/project';
import { SavingsGroupBeneficiary } from './models/savings-group-beneficiary';
import { SavingsGroupMonitoring } from './models/savings-group-monitoring';

import adapter from './adapter';
import { Action } from './models/action';
import { ActionType } from './models/action-type';
import { Beneficiary } from './models/beneficiary';
import { BeneficiaryInitiative } from './models/beneficiary-initiative';
import { Cycle } from './models/cycle';
import { Initiative } from './models/initiative';
import { Issue } from './models/issue';
import { Location } from './models/location';
import { ParticipantAction } from './models/participant-action';
import { SavingsGroup } from './models/savings-group';
import { TimeEntry } from './models/time-entry';
import { ActionPlan } from './models/action-plan';
import { User } from './models/user';
import { ActionPlanReport } from './models/action-plan-report';
import { SavingsGroupBeneficiaryMonitoring } from './models/savings-group-beneficiary-monitoring';
// Then, make a Watermelon database from it!
export const database = new Database({
  adapter,
  modelClasses: [
    Location,
    User,
    Initiative,
    Beneficiary,
    BeneficiaryInitiative,
    Action,
    ActionType,
    ParticipantAction,
    Project,
    Issue,
    TimeEntry,
    SavingsGroup,
    SavingsGroupMonitoring,
    SavingsGroupBeneficiary,
    Cycle,
    ActionPlan,
    ActionPlanReport,
    SavingsGroupBeneficiaryMonitoring,
    MonitoringVisit,
  ],
});

export const resetDatabase = async () =>
  await database.write(async () => await database.unsafeResetDatabase());

if (__DEV__ && !__TEST__) {
  const {
    connectDatabases,
    WatermelonDB,
  } = require('react-native-flipper-databases');
  connectDatabases([
    new WatermelonDB(database), // Pass in database definition
  ]);
}
