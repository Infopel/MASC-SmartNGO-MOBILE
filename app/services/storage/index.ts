export * from "./database"
// export * from "./queries"
export * from "./models"
export * from "./migrations"