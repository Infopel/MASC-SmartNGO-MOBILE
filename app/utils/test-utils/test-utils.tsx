import { render } from "@testing-library/react-native"
import React from "react"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProviders({ children }: AllTheProvidersProps) {
  return (
    <ThemeProvider>{children}</ThemeProvider>
  )
}

function customRender<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
  return render(ui, { wrapper: AllTheProviders, ...options })
}

// re-export everything
export * from "@testing-library/react-native"
// override render method
export { customRender as render }

