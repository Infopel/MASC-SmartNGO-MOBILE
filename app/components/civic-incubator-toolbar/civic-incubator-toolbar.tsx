import { Heading } from "components/heading/heading"
import { useToolbarColor } from "components/hooks"
import { Icon } from "components/icon/icon"
import { InitiativeToolbarActions } from "components/initiative-toolbar-actions/initiative-toolbar-actions"
import { HStack, IBoxProps, IconButton, VStack } from "native-base"
import { goBack } from "navigators/navigation-utilities"
import * as React from "react"
import { Platform } from "react-native"
import { IInitiative } from "storage/queries/initiative"
import { DateUtils } from "utils/DateUitls"
import { TextUtils } from "utils/TextUtils"

export interface CivicIncubatorToolbarProps extends IBoxProps {
  incubator: IInitiative
}

/**
 * Describe your component here
 */
export function CivicIncubatorToolbar({ incubator, ...rest }: CivicIncubatorToolbarProps) {


  const isIOS = Platform.OS === "ios"
  const { bg, border } = useToolbarColor()
  return (
    <HStack {...rest} bg={bg} py='3' px='1' alignItems={'flex-start'} borderColor={border}>
      <IconButton p={1} onPress={goBack} testID="back" colorScheme={'text'} icon={<Icon size={'xl'} name='arrow-back' type='material' />} />
      <VStack space='3' flex={1}>

        <Heading fontSize={'xl'} fontWeight='medium' textAlign={isIOS ? "center" : undefined} noOfLines={1}
        //@ts-ignore
        >{incubator.name}</Heading>
        <HStack alignItems={'center'} space='1'>
          <Icon name='place' type='material' size='sm' />
          <Heading fontSize={'sm'} fontWeight='light' noOfLines={1}
          //@ts-ignore
          >{TextUtils.getIncubatorLocation(incubator.province, incubator.district)}</Heading>
        </HStack>
        <HStack alignItems={'center'} space='1'>
          <Icon name='calendar-today' type='material' size='sm' />
          <Heading fontSize={'sm'} fontWeight='light' noOfLines={1}
          //@ts-ignore
          >{DateUtils.formatToShortDate(incubator.date)}</Heading>
        </HStack>
      </VStack>
      <InitiativeToolbarActions initiativeId={incubator.id}/>
    </HStack>
  )
}
