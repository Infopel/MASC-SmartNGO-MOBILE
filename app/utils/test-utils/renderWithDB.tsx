jest.mock("storage/adapter.ts")
import DatabaseProvider from "@nozbe/watermelondb/DatabaseProvider"
import { render } from "@testing-library/react-native"
import { default as React } from "react"
import { database } from "storage/database"
import { NativeBaseProvider as ThemeProvider } from "theme"
import { AllTheProvidersProps, RenderOptions } from "./render"

function AllTheProvidersWithDB({ children }: AllTheProvidersProps) {
    return (
        <DatabaseProvider database={database}>
            <ThemeProvider>{children}</ThemeProvider>
        </DatabaseProvider>
    )
}

export function renderWithDB<T extends any>(ui: React.ReactElement<T>, options?: RenderOptions) {
    return render(ui, { wrapper: AllTheProvidersWithDB, ...options })
}


export class DiagnosticError extends Error {
    name: string = "Diagnostic error"
  }
  