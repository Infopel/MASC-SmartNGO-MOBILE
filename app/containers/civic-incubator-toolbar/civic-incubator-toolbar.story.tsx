import * as React from "react"
import { storiesOf } from "@storybook/react-native"
import { StoryScreen, Story, UseCase } from "../../../storybook/views"
import { color } from "../../theme"
import { CivicIncubatorToolbar } from "./civic-incubator-toolbar.component"

storiesOf("CivicIncubatorToolbar", module)
  .addDecorator((fn) => <StoryScreen>{fn()}</StoryScreen>)
  .add("Style Presets", () => (
    <Story>
      <UseCase text="Primary" usage="The primary.">
        <CivicIncubatorToolbar style={{ backgroundColor: color.error }} />
      </UseCase>
    </Story>
  ))
