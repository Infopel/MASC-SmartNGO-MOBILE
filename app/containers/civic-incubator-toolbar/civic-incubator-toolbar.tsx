import Database from "@nozbe/watermelondb/Database"
import { withDatabase } from "@nozbe/watermelondb/DatabaseProvider"
import withObservables from "@nozbe/with-observables"
import { CivicIncubatorToolbar } from "components/civic-incubator-toolbar/civic-incubator-toolbar"
import { observer } from "mobx-react-lite"
import { IBoxProps } from "native-base"
import { PrimaryStackHeaderProps } from "navigators/navigators"
import * as React from "react"
import { findInitiativeById, IInitiative } from "storage/queries/initiative"

interface ICivicIncubatorToolbarDatabaseQuery extends PrimaryStackHeaderProps<'civic-incubator-details'> {
  database: Database

}

export interface CivicIncubatorToolbarProps extends IBoxProps, ICivicIncubatorToolbarDatabaseQuery {
  incubator: IInitiative
  incubatorId: string
}

/**
 * Describe your component here
 */
export const _CivicIncubatorToolbarContainer = observer(function ({ incubator }: CivicIncubatorToolbarProps) {

  return (
    <CivicIncubatorToolbar incubator={incubator} />
  )
})


export const CivicIncubatorToolbarContainer = withDatabase(withObservables(['route', 'database'], ({ database, route: { params: { incubatorId } } }: ICivicIncubatorToolbarDatabaseQuery) => ({ incubator: findInitiativeById(incubatorId, database) }))(_CivicIncubatorToolbarContainer))
