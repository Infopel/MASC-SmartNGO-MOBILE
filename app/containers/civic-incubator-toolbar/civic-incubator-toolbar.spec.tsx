import { fireEvent, render } from "utils/test-utils/test-utils"
import * as React from "react"
import { CivicIncubatorToolbarContainer } from "./civic-incubator-toolbar"
import { PrimaryStackHeaderProps } from "navigators/navigators"
import { database } from "storage/database"
import { renderWithDB } from "utils/test-utils/renderWithDB"

describe("<CivicIncubatorToolbarContainer>", () => {
  it("render", async function () {
    const nav = {
      route: {
        key: "civic-incubator-toolbar-key", params: { incubatorId: "50" }, name: "civic-incubator-details", path: "/civic-incubator"
      }
    } as PrimaryStackHeaderProps<'civic-incubator-details'>
    const component = <CivicIncubatorToolbarContainer {...nav} database={database}></CivicIncubatorToolbarContainer>

    expect(renderWithDB(component).toJSON()).toMatchSnapshot()
  })
})
