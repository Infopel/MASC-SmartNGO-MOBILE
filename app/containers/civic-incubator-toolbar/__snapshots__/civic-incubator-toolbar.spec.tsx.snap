// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CivicIncubatorToolbarContainer> render 1`] = `null`;
